aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -8,7 +8,36 @@
 
 void runMPSGraph(MPSStream* mpsStream, MPSGraph* mpsGraph, NSDictionary* feeds,
                  NSDictionary* results, bool disableTypeInference) {
-  mpsStream->executeMPSGraph(mpsGraph, feeds, results, SyncType::COMMIT_ADAPTIVE, disableTypeInference);
+  mpsStream->executeMPSGraph(mpsGraph, feeds, results, SyncType::COMMIT_ADAPTIVE);
+}
+
+// this should be merged into runMPSGraph() with new arg "disableTypeInference" for executables
+void runMPSGraphExecutable(MPSStream *mpsStream, MPSCachedGraph *cachedGraph, NSDictionary *feeds, NSDictionary *results, MPSGraphTensor* alphaTensor) {
```

Comment:
This is a generic function and Binary Op-specific arguments like alphaTensor shouldn't be in here.
As discussed, please make the alphaTensor unranked in its origin (i.e., Binary ops) when type inference is disabled).

Reply:
Thanks, updated.

Reply:
You don't need to pass alphaTensor here. it should already be in the feeds.

Reply:
Yea, I wanted to get rid completely of the `runMPSGraphExecutable` function - and just have a generic function with `runMPSGraph`. I've updated that now